solidity/test/rewards/TestECDSARewards.js

```diff
@@ -0,0 +1,351 @@
+
+      await timeJumpToEndOfInterval(0)
+
+      const keepTerminatedAddress = await keepFactory.getKeepAtIndex(0)
+      const keepTerminated = await BondedECDSAKeepStub.at(keepTerminatedAddress)
+      await keepTerminated.publicMarkAsTerminated()
+
+      const keepAddress = await keepFactory.getKeepAtIndex(1)
+      const keep = await BondedECDSAKeepStub.at(keepAddress)
+      await keep.publicMarkAsClosed()
+
+      await expectRevert(
+        rewardsContract.receiveReward(keepTerminatedAddress),
+        "Keep is not closed"
+      )
+
+      await rewardsContract.receiveReward(keepAddress)
+      // Only the second keep was properly closed.
+      await assertKeepBalanceOfBeneficiaries(expectedBeneficiaryBalance)
+
+      // the remaining 5346000 stays in unallocated rewards but the fact
+      // one keep was terminated needs to be reported to recalculate the 
+      // unallocated amount
+      let unallocated = await rewardsContract.unallocatedRewards()
+      let unallocatedInKeep = unallocated.div(tokenDecimalMultiplier)
+      expect(unallocatedInKeep).to.eq.BN(174636000) // 178200000 - (2 * 1782000)
+
+      await rewardsContract.reportTermination(keepTerminatedAddress)
+
+      unallocated = await rewardsContract.unallocatedRewards()
+      unallocatedInKeep = unallocated.div(tokenDecimalMultiplier)
+      expect(unallocatedInKeep).to.eq.BN(176418000) // 178200000 - 1782000
+    })
+
+    it("should correctly distribute rewards between beneficiaries", async () => {
+      await createKeeps(8, firstKeepCreationTimestamp)
+      // reward for the first interval: 7128000 KEEP
+      // keeps created: 8 => 891000 KEEP per keep
+      // member receives: 891000 / 3 = 297000 (3 signers per keep)
+      const expectedBeneficiaryBalance = new BN(297000)
+
+      await timeJumpToEndOfInterval(0)
+
+      let keepAddress = await keepFactory.getKeepAtIndex(0)
+      let keep = await BondedECDSAKeepStub.at(keepAddress)
+      await keep.publicMarkAsClosed()
+
+      await rewardsContract.receiveReward(keepAddress)
+
+      await assertKeepBalanceOfBeneficiaries(expectedBeneficiaryBalance)
+
+      // verify second keep in this interval
+      keepAddress = await keepFactory.getKeepAtIndex(1)
+      keep = await BondedECDSAKeepStub.at(keepAddress)
+      await keep.publicMarkAsClosed()
+
+      await rewardsContract.receiveReward(keepAddress)
+
+      // 297000 * 2 = 594000
+      await assertKeepBalanceOfBeneficiaries(expectedBeneficiaryBalance.muln(2))
+    })
+  })
+
+  async function assertKeepBalanceOfBeneficiaries(expectedBalance) {
+    for (let i = 0; i < keepMembers.length; i++) {
+      const actualBalance = (
+        await keepToken.balanceOf(keepMembers[i].beneficiary)
+      ).div(tokenDecimalMultiplier)
+
+      expect(actualBalance).to.gte.BN(expectedBalance.subn(precision))
+      expect(actualBalance).to.lte.BN(expectedBalance.addn(precision))
+    }
+  }
+
+  async function initializeNewFactory() {
```

Comment:
Most of the code below is identical with what's in `TestECDSABackportRewards.js`. Can we extract them into some file in `rewards` dir?